client/dashboard/components/navigation-blocker/navigation-blocker-registry.tsx

```diff
@@ -0,0 +1,23 @@
+import { useBlocker } from '@tanstack/react-router';
+import { useSyncExternalStore } from 'react';
+import { navigationStore } from './navigation-blocker-store';
+
+export function NavigationBlockerRegistry() {
```

Comment:
Nit: given that it's stateful and not really a reusable component, maybe it's more appropriate to put everything in `client/dashboard/app/*` instead 🤔

Reply:
Done by https://github.com/Automattic/wp-calypso/pull/107859/commits/533e161bd4eb7246c4ac7f72580fd7a973406cd0.